src/lighteval/models/model_input.py

```diff
@@ -0,0 +1,154 @@
+            top_k=config_dict["generation"].get("top_k", None),
+            min_p=config_dict["generation"].get("min_p", None),
+            top_p=config_dict["generation"].get("top_p", None),
+            truncate_prompt=config_dict["generation"].get("truncate_prompt", None),
+        )
+
+    def to_vllm_openai_dict(self) -> dict:
+        """Selects relevant generation and sampling parameters for vllm and openai models.
+        Doc: https://docs.vllm.ai/en/v0.5.5/dev/sampling_params.html
+
+        Returns:
+            dict: The parameters to create a vllm.SamplingParams or just provide OpenAI params as such in the model config.
+        """
+        # Task specific sampling params to set in model: n, best_of, use_beam_search
+        # Generation specific params to set in model: logprobs, prompt_logprobs
+        args = {
+            "presence_penalty": self.presence_penalty,
+            "frequency_penalty": self.frequency_penalty,
+            "repetition_penalty": self.repetition_penalty,
+            "temperature": self.temperature,
+            "top_p": self.top_p,
+            "top_k": self.top_k,
+            "min_p": self.min_p,
+            "seed": self.seed,
+            "length_penalty": self.length_penalty,
+            "early_stopping": self.early_stopping,
+            "stop": self.stop_tokens,
+            "max_tokens": self.max_new_tokens,
+            "min_tokens": self.min_new_tokens,
+            "truncate_prompt_tokens": self.truncate_prompt,
+        }
+        return {k: v for k, v in args.items() if v is not None}
+
+    def to_transformers_dict(self) -> dict:
+        """Selects relevant generation and sampling parameters for transformers models.
+        Doc: https://huggingface.co/docs/transformers/v4.46.3/en/main_classes/text_generation#transformers.GenerationConfig
+
+        Note: We actually don't use the GenerationConfig object itself because it has a huge number of parameters automatically
+        initialized, to a config which slows down evals insanely.
+
+        Returns:
+            dict: The parameters to create a transformers.GenerationConfig in the model config.
+        """
+        # Task specific sampling params to set in model: do_sample, num_return_sequences, num_beans
+        args = {
+            "max_new_tokens": self.max_new_tokens,
+            "min_new_tokens": self.min_new_tokens,
+            "early_stopping": self.early_stopping or False,
+            "stop_strings": self.stop_tokens,
+            "temperature": self.temperature,
+            "top_k": self.top_k,
+            "top_p": self.top_p,
+            "min_p": self.min_p,
+            "repetition_penalty": self.repetition_penalty,
+            "length_penalty": self.length_penalty,
+            "output_scores": True,
+            "return_dict_in_generate": True,
+        }
+        return {k: v for k, v in args.items() if v is not None}
+
+    def to_tgi_inferenceendpoint_dict(self) -> dict:
```

Comment:
Maybe renaming to `to_tgi_endpoint_dict`?
- TGI has the word `inference` included

Reply:
it's because it's either tgi or inference endpoints. Maybe tgi_ie?